be/src/olap/tablet_manager.cpp

```diff
@@ -1490,9 +1490,13 @@ void TabletManager::get_tablets_distribution_on_different_disks(
                     std::map<int64_t, std::map<DataDir*, int64_t>> &tablets_num_on_disk,
                     std::map<int64_t, std::map<DataDir*, std::vector<TabletSize>>> &tablets_info_on_disk) {
     std::vector<DataDir*> data_dirs = StorageEngine::instance()->get_stores();
-    ReadLock rlock(&_partition_tablet_map_lock);
-    std::map<int64_t, std::set<TabletInfo>>::iterator partition_iter = _partition_tablet_map.begin();
-    for (; partition_iter != _partition_tablet_map.end(); partition_iter++) {
+    std::map<int64_t, std::set<TabletInfo>> partition_tablet_map;
+    {
+        ReadLock rlock(&_partition_tablet_map_lock);
```

Comment:
Could you please to add comment to explain why we need lock and copy the `_partition_tablet_map`?

Reply:
> Could you please to add comment to explain why we need lock and copy the `_partition_tablet_map`?

Done.